node/core/av-store/src/lib.rs

```diff
@@ -966,54 +980,57 @@ fn store_available_data(
 }
 
 #[tracing::instrument(level = "trace", skip(subsystem), fields(subsystem = LOG_TARGET))]
-fn store_chunk(
+fn store_chunks(
 	subsystem: &mut AvailabilityStoreSubsystem,
 	candidate_hash: &CandidateHash,
-	_n_validators: u32,
-	chunk: ErasureChunk,
+	chunks: &[ErasureChunk],
 	block_number: BlockNumber,
 ) -> Result<(), Error> {
-	let _timer = subsystem.metrics.time_store_chunk();
+	let _timer = subsystem.metrics.time_store_chunks();
 
 	let mut tx = DBTransaction::new();
+	let mut chunk_pruning = chunk_pruning(&subsystem.inner).unwrap_or_default();
 
-	let dbkey = erasure_chunk_key(candidate_hash, chunk.index);
+	for chunk in chunks {
+		subsystem.chunks_cache.entry(*candidate_hash).or_default().insert(chunk.index, chunk.clone());
```

Comment:
This has to be moved at the end of loop and changed to
```suggestion
		subsystem.chunks_cache.entry(*candidate_hash).or_default().extend(chunks.into_iter().map(|c| (c.index, c));
```